library/src/options.cxx

```diff
@@ -29,10 +32,16 @@ class options::internals
   };
 
   template<typename T>
-  void init(const std::string& name, const T& value)
+  void init(const std::string& name, const T& value, [[maybe_unused]] bool deprecated = false)
   {
     static_assert(IsTypeValid<T, OptionVariant>::value);
     this->Options[name] = value;
+#ifndef F3D_NO_DEPRECATED
+    if (deprecated)
```

Comment:
Is the `deprecated` flag useful here? It seems like it's already checked by the build time condition.

Reply:
the deprecated flag is here to add the option to the deprecated options set.
the F3D_NO_DEPRECATED is here to add this block of code or not.

Reply:
Right, I misread the code.  
Then I think it would be better (easier to read, and more performant) to just create a function in `f3d::internals` called like:
```cpp
bool IsDeprecated(const std::string& name)
{
  // compile time list of deprecated options
  constexpr std::string_view deprecated[] = { "render.background.hdri" };

  auto it = std::find(std::begin(deprecated), std::end(deprecated), name);
  return it != std::end(deprecated);
}
```
And call `internals->IsDeprecated(name)` instead of `this->DeprecatedOptions.count(name) > 0`

Reply:
I suppose this is compile time find like this ?

Reply:
the find is not compile time because `name` argument is not known at compile time.  
But the list is so the compiler can optimize it out if it's relevant. And no need to store the list in heap.